src/app/main.cpp

```diff
@@ -326,15 +326,20 @@ int main(int argc, char **argv) {
         frontendManager = new FrontendManager(pluginDirs);
         extensionManager = new ExtensionManager(pluginDirs);
         extensionManager->reloadExtensions();
-        hotkeyManager = new HotkeyManager;
-        if ( parser.isSet("hotkey") ) {
-            QString hotkey = parser.value("hotkey");
-            if ( !hotkeyManager->registerHotkey(hotkey) )
-                qFatal("Failed to set hotkey to %s.", hotkey.toLocal8Bit().constData());
-        } else if ( settings.contains("hotkey") ) {
-            QString hotkey = settings.value("hotkey").toString();
-            if ( !hotkeyManager->registerHotkey(hotkey) )
-                qFatal("Failed to set hotkey to %s.", hotkey.toLocal8Bit().constData());
+
+        if ( !QGuiApplication::platformName().contains("wayland") )
```

Comment:
I thought about whether this is the right thing in all wayland environments, and I think it is. I'm not aware of a wayland protocol for requesting a hotkey from the compositor, and if there is it can be supported in a later patch.